src/stc/PlatWX.cpp

```diff
@@ -44,7 +44,7 @@
 #include "wx/stc/stc.h"
 #include "wx/stc/private.h"
 
-#if wxUSE_GRAPHICS_DIRECT2D
+#if defined(__WXMSW__) && wxUSE_GRAPHICS_DIRECT2D
```

Comment:
Could be worth introducing a special symbol for this (just in this file, not `setup.h`) to avoid repeating the same condition, e.g. `HAVE_DIRECTWRITE_TECHNOLOGY`?